server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1352,40 +1351,45 @@ private void cleanupStaleBlobs(
      *     <li>Deleting stale indices</li>
      *     <li>Deleting unreferenced root level blobs</li>
      * </ul>
-     * @param repositoryStateId     Current repository state id
-     * @param repositoryMetaVersion version of the updated repository metadata to write
+     * @param originalRepositoryDataGeneration     Current repository state id
+     * @param repositoryFormatIndexVersion version of the updated repository metadata to write
      * @param listener              Listener to complete when done
      */
-    public void cleanup(long repositoryStateId, IndexVersion repositoryMetaVersion, ActionListener<RepositoryCleanupResult> listener) {
+    public void cleanup(
+        long originalRepositoryDataGeneration,
+        IndexVersion repositoryFormatIndexVersion,
+        ActionListener<RepositoryCleanupResult> listener
+    ) {
         try {
             if (isReadOnly()) {
                 throw new RepositoryException(metadata.name(), "cannot run cleanup on readonly repository");
             }
-            Map<String, BlobMetadata> rootBlobs = blobContainer().listBlobs(OperationPurpose.SNAPSHOT);
-            final RepositoryData repositoryData = safeRepositoryData(repositoryStateId, rootBlobs);
-            final Map<String, BlobContainer> foundIndices = blobStore().blobContainer(indicesPath()).children(OperationPurpose.SNAPSHOT);
-            final Set<String> survivingIndexIds = repositoryData.getIndices()
+            Map<String, BlobMetadata> originalRootBlobs = blobContainer().listBlobs(OperationPurpose.SNAPSHOT);
+            final RepositoryData originalRepositoryData = safeRepositoryData(originalRepositoryDataGeneration, originalRootBlobs);
```

Comment:
The `safeRepositoryData` method can also use a rename for its first parameter. Currently it is `repositoryStateId`. But `repositoryDataGeneration` feels more inline with the proposed changes.

Reply:
Yes, I haven't quite decided what to do with `safeRepositoryData` yet but that renaming definitely fits.